### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by ahzamir**
### In this pull request I have added some classes and changes to existing classes to set associations between classes according to these requirements:

- I created a class Classroom with the following:
  - @label instance variable, initialized in the constructor.
  - Setter and getter for @label (attr_accessor).
- I created the has-many/belongs-to relationship between Classroom and Student. - - The following has been implemented:
  - I created the has-many side (a classroom has many students).
  - I created the belongs-to side (a student belongs to a classroom).
  - When adding a student to a classroom it also sets the classroom for the student.
  - When setting the classroom for a student it also adds it to the classrooms' students.
- I created a class Book with the following:
  - @title and @author instance variables, initialized in the constructor.
  - Setters and getters for instance variables (attr_accessor).
- I created a class Rental with the following:
  - @date instance variable, initialized in the constructor.
  - Setter and getter for @date (attr_accessor).
- I created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. 
- The following has been implemented:
  - I created the has-many side of Book and Rental (a book has many rentals).
  - I created the belongs-to side of Rental and Book (a rental belongs to a book).
  - I created the has-many side of Person and Rental (a person has many rentals).
  - I created the belongs-to side of Rental and Person (a rental belongs to a person).
  - I modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -7,11 +8,12 @@ def initialize(age, name = 'Unknown', parent_permission: true)
...
+    @rentals = []
   end
 
   attr_reader :id
 
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :parent_permission, :rentals
 
   def of_age?
     @age >= 18
```
Comment (Deepakdanger) [inline person.rb:19]: - [x] Here method `of_age?` is defined but on line 24 you have used `is_of_age? `. Kindly change to`of_age?` on line 24 inside the `can_use_services?` method.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @ahzamir, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (ahzamir) [reply person.rb:19]: @Deepakdanger  Thank you very much for your kind review. Yeah, it was showing linters so I changed it to `of_age?` and forgot to change that one too. I will change it now thanks 👍😊

**Review by KDlamini — APPROVED** [review]
### Hi @ahzamir 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  Relationships and associations implemented correctly
✔️  All relevant Getters and Settlers included
✔️  Good PR title and description
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahzamir** [comment]
@Deepakdanger  @KDlamini Thank you very much for your kind review and feedback. 😊👍

**Pull request merged**